### ISSUE TITLE
Allow passing the initial value to realize()

### ISSUE THREAD

**Issue by ancarda** [ancarda/functional#2]
Operations can be generated once, and executed multiple times. Usually, one would want to replace the input value, and realize over and over. This could be much faster if the input doesn't need to be specified as that causes a whole clone, and may not be that cheap.

Realize should take an optional parameter: `realize($init = null)` which if specified (not null), will be used over what was specified with `input()`. If null is desired, and there's already an initial value, it'll need to be replaced with `input(null)->realize()`.

Thanks to @Zegnat for suggesting this.

**Comment by Zegnat** [comment]
In addition I would suggest aliasing `realize()` as [`__invoke()`](https://secure.php.net/manual/en/language.oop5.magic.php#object.invoke):

```php
    public function __invoke($init = null)
    {
        return $this->realize($init);
    }
```

This will allow sort of a dynamic function building. Where an operation that I want to be used multiple times can be called directly:

```php
$uniquelyFlatten = (new Operation())->flatten()->deduplicate();
$values = $uniquelyFlatten([[1, 2, 3], [3, 4, 5]]);
```

This will also make the object callable for other purposes, such as passing it to other PHP functions that expect callables.

**Comment by ancarda** [comment]
Fixed in 7922c5d19c2ab0c2ad3cf9115d24adbb537972d8 and aac9f27814c26a07bad11f43f83f07b542fddb9f
